components/faction/commons/faction.lua

```diff
@@ -66,17 +83,20 @@ function Faction.read(faction, options)
 	options = options or {}
 
 	faction = faction:lower()
-	return Faction.isValid(faction) and faction
-		or byLowerName[faction]
-		or (options.alias ~= false and Faction.aliases[faction])
-		or nil
+	return Faction.isValid(faction, options.game) and faction
+		or (String.isNotEmpty(options.game) and (byLowerName[options.game] or {})[faction] or byLowerName[faction])
+		or (
+			options.alias ~= false
+			and (String.isNotEmpty(options.game) and (Faction.aliases[options.game] or {})[faction]
+			or Faction.aliases[faction])
+		) or nil
```

Comment:
logic for the alias stuff seems faulty
probably need to move the `options.alias ~= false and` to before the `(`

Reply:
I think the indentation is just bad, the `or Faction.aliases` is still inside the RHS of the `options.alias ~= false and`

Reply:
ah